rules.go

```diff
@@ -152,13 +152,29 @@ func googleSearchType(ref Referrer) GoogleSearchType {
 		return NotGoogleSearch
 	}
 
-	if strings.HasPrefix(ref.Path, "/aclk") || strings.HasPrefix(ref.Path, "/pagead/aclk") {
+	if strings.HasPrefix(ref.Path, "/aclk") || strings.HasPrefix(ref.Path, "/pagead/aclk") || hasGCLID(ref.URL) {
 		return Adwords
 	}
 
 	return OrganicSearch
 }
 
+func hasGCLID(rawURL string) bool {
+	u, err := url.Parse(rawURL)
```

Comment:
We should probably avoid parsing the URL again. We can just pass it in from `ParseWith()`.